### ISSUE TITLE
Strawberry Shake - Multiple query types

### ISSUE THREAD

**Issue by marian-margeta** [ChilliCream/hotchocolate#2008]
**Describe the bug**
I have a problem with generating client when more than one query type is declared

**To Reproduce**

Let's say I have those files in my client project:

berry.json
```json
{
  "Schemas": [
    {
      "Name": "MyObject",
      "Type": "http",
      "File": "Example.graphql"
    }
  ],
  "ClientName": "MyObjectClient"
}
```

Example.graphql
```graphql
schema {
  query: Query
}

type Query {
  getMyObjects: [MyObject!]!
  getOtherObjects: [OtherObject!]!
}

type MyObject {
  id: Long!
  text: String
}

type OtherObject {
  id: Long!
  text: String
}
```

Now I want to declare 2 queries:
```graphql
query get01 {
  getMyObjects {
    text
  }
}
```
and
```graphql
query get02 {
  getOtherObjects {
    text
  }
}
```
If I put those queries to separate files (one query per .graphql file) I get this error from Strawberry:
```csharp
Unhandled Exception: System.InvalidOperationException: The file name `C:\test\Client\Generated\IMyObjectClient.cs` was already used.
1>   at StrawberryShake.Generators.DirectoryFileHandler.ExecuteGeneratorAsync(GeneratorTask task, ITypeLookup typeLookup, ISet`1 usedNames) in D:\a\1\s\src\StrawberryShake\CodeGeneration\src\Generators\DirectoryFileHandler.cs:line 69
1>   at StrawberryShake.Generators.DirectoryFileHandler.WriteAllAsync(ITypeLookup typeLookup) in D:\a\1\s\src\StrawberryShake\CodeGeneration\src\Generators\DirectoryFileHandler.cs:line 58
1>   at StrawberryShake.Generators.ClientGenerator.BuildAsync() in D:\a\1\s\src\StrawberryShake\CodeGeneration\src\Generators\ClientGenerator.cs:line 362
1>   at StrawberryShake.Tools.GenerateCommandHandler.Compile(GenerateCommandContext context, String path, Configuration config, ClientGenerator generator, IReadOnlyList`1 documents, ICollection`1 errors) in D:\a\1\s\src\StrawberryShake\Tooling\src\dotnet-graphql\GenerateCommandHandler.cs:line 84
1>   at StrawberryShake.Tools.CompileCommandHandlerBase`2.Compile(TCtx context, String path, Configuration configuration) in D:\a\1\s\src\StrawberryShake\Tooling\src\dotnet-graphql\CompileCommandHandlerBase.cs:line 121
1>   at StrawberryShake.Tools.CompileCommandHandlerBase`2.Compile(TCtx context, String path) in D:\a\1\s\src\StrawberryShake\Tooling\src\dotnet-graphql\CompileCommandHandlerBase.cs:line 91
1>   at StrawberryShake.Tools.CompileCommandHandlerBase`2.ExecuteAsync(TCtx context) in D:\a\1\s\src\StrawberryShake\Tooling\src\dotnet-graphql\CompileCommandHandlerBase.cs:line 70
1>   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync(String[] args, CancellationToken cancellationToken)
1>   at StrawberryShake.Tools.Program.<Main>(String[] args)
```

If all the queries are defined in one file, everything seems to work. Strawberry generates the code correctly and `IMyObjectClient` contains 2 methods: `Get01Async` and `Get02Async`. However if I call the method `Get01Async()`, GraphQL server gets the json file where all the queries are included (get01 as well as get02), so GraphQL server executes both queries:
```json
{
    "id": "c651e5a7a6e0100a2e57effb17d7675c",
    "query": "query get01 { getMyObjects { __typename text } } query get02 { getOtherObjects { __typename text } }",
    "operationName": "get01"
}
```

**Expected behavior**
1. I would expect that it does not matter whether the queries are in one file or multiple files 
2. If I call the method for one query (get01) I don't want all the queries to be executed. 

**Desktop (please complete the following information):**
 - OS: `Windows 10`
 - Version: `StrawberryShake:11.0.0-preview.133`

**Comment by iskandersierra** [comment]
Was this removed from the milestone?
This issue involves two subjects:
- Error generating multiple Queries files into the single interface `IMyApiClient`
- Optimize sending the relevant Query file instead of the merge.

I think the first subject can easily be solved using partial interfaces. Currently the interface IMyApiClient is already partial, we just need to use a unique filename for each Query file, like:
- `IMyApiClient.Queries1.cs`
- `IMyApiClient.Queries2.cs`

Is there some effort being put into solving this issue? I can work on a quick fix like this for the first issue.

I also need multiple queries, because I'm generating a scaffolding from a model and each queries file is generated at different moments, so having different files is the best approach for my case.

**Comment by michaelstaib** [comment]
Strawberry Shake release is slated for January.

**Comment by michaelstaib** [comment]
This issue is now resolved and is fixed with the newest preview 11.1.0-preview.58 ... 

look at the demo:
https://github.com/michaelstaib/PublicSpeaking/tree/master/channel_9_live/Demo_1

It has multiple query files and fragments can be spread over many files as well.

We will follow up with documentation and blogs on this over the weekend.
